commons/src/main/java/com/powsybl/commons/parameters/ConfiguredParameter.java

```diff
@@ -0,0 +1,94 @@
+        super(parameter.getNames(),
+                parameter.getType(),
+                parameter.getDescription(),
+                parameter.getDefaultValue(),
+                parameter.getPossibleValues(),
+                parameter.getScope(),
+                parameter.getCategoryKey());
+        this.configuredValue = checkDefaultValue(parameter.getType(), configuredValue);
+    }
+
+    @Override
+    public ParameterSourceType getParameterSourceType() {
+        return ParameterSourceType.CONFIGURED_PARAMETER;
+    }
+
+    public Object getConfiguredValue() {
+        return configuredValue;
+    }
+
+    /**
+     * Loads a list of parameters and dynamically assigns default values from the provided configuration {@link ParameterDefaultValueConfig}.
+     * If the defaultValueConfig is null, the method returns the original list of parameters.
+     * Otherwise, each parameter is processed with a potential dynamic default value derived from the configuration.
+     *
+     * @param parameters        The collection of parameters to be processed.
+     * @param prefix            The prefix used to fetch values from the defaultValueConfig.
+     * @param defaultValueConfig The configuration containing default values for parameters.
+     * @return A list of parameters with dynamically assigned default values.
+     */
+    public static List<Parameter> load(Collection<Parameter> parameters, String prefix, ParameterDefaultValueConfig defaultValueConfig) {
+        return defaultValueConfig == null
+                ? ImmutableList.copyOf(parameters)
+                : parameters.stream()
+                .map(param -> processParameters(defaultValueConfig.getValue(prefix, param), param))
+                .toList();
+    }
+
+    private static Parameter processParameters(Object configuredValue, Parameter param) {
+        if (configuredValue != param.getDefaultValue()) {
+            return new ConfiguredParameter(param, configuredValue);
+        }
+        return param;
+    }
+
+    /**
+     * Loads a list of parameters and dynamically assigns default values from the provided configuration {@link ModuleConfig}.
+     * If the moduleConfig is null, the method returns the original list of parameters.
+     * Otherwise, each parameter is processed with a potential dynamic default value derived from the configuration.
+     *
+     * @param parameters  The collection of parameters to be processed.
+     * @param moduleConfig The module configuration containing default values for parameters.
+     * @return A list of parameters with dynamically assigned default values.
+     */
+    public static List<Parameter> load(Collection<Parameter> parameters, ModuleConfig moduleConfig) {
+        return moduleConfig == null
+                ? ImmutableList.copyOf(parameters)
+                : parameters.stream()
+                .map(param -> processParameters(getValueFromModuleConfig(param, moduleConfig), param))
```

Comment:
This may throw an exception if the value in config for a parameter is invalid regarding the parameter (wrong type or not among its acceptable values).
Something must be done to prevent the whole "load" method to fail...